lib/vector-core/src/enrichment_table/enrichment_tables.rs

```diff
@@ -0,0 +1,286 @@
+use super::EnrichmentTable;
+use arc_swap::ArcSwap;
+#[cfg(feature = "vrl")]
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use std::{collections::HashMap, sync::Mutex};
+
+/// The Enrichment Tables manages the collection of `EnrichmentTable`s loaded into Vector.
+/// Enrichment Tables go through two stages.
+///
+/// ## 1. Writing
+///
+/// The  tables are loaded. There are two elements that need loading. The first is the actual data.
+/// This is loaded at config load time, the actual loading is performed by the implementation of
+/// the `EnrichmentTable` trait. Next, the tables are passed through Vectors `Transform` components,
+/// particularly the `Remap` transform. These Transforms are able to determine which fields we will
+/// want to lookup whilst Vector is running. They can notify the tables of these fields no that the
+/// data can be indexed.
+///
+/// During this phase, the data needs to be mutated and shared around a number of potential
+/// different components. Also, performance is not absolutely critical at this phase. Thus, the
```

Comment:
Might be good to add _why_ performance isn't critical (i.e. because it happens at startup).